### PR TITLE
Added add-library instructions

### PR DESCRIPTION
**Pull request by jducoeur**
As discussed on Gitter.

### CODE REVIEW

**Comment by gzm0** [comment]
:-( Please reference the ticket I created just for you: #144

File: index.md [hunk 1]
```diff
@@ -64,6 +64,12 @@ Some of them, like `scala-js-dom` and `scala-js-jquery`, are specific to Javascr
 Some others, such as `scala-async`, are pure-macro projects, and thus work with Scala.js out of the box.
 Most of them, though, are cross-compiling over JVM and JS, and thus have separate artifacts to run on each platform.
 
+If you have a library that would be useful to the Scala.js community, please add it to this list! 
+[This link](https://github.com/scala-js/scala-js-website/edit/master/index.md) takes you directly to the
```
Comment (gzm0) [inline index.md]: Links saying "click here" are in general considered bad practice. Just make "add it to this list" a link and omit the next sentence entirely.

File: index.md [hunk 2]
```diff
@@ -64,6 +64,12 @@ Some of them, like `scala-js-dom` and `scala-js-jquery`, are specific to Javascr
 Some others, such as `scala-async`, are pure-macro projects, and thus work with Scala.js out of the box.
 Most of them, though, are cross-compiling over JVM and JS, and thus have separate artifacts to run on each platform.
 
+If you have a library that would be useful to the Scala.js community, please add it to this list! 
+[This link](https://github.com/scala-js/scala-js-website/edit/master/index.md) takes you directly to the 
+editor for this page in GitHub. Add a new entry for the library under the appropriate category below (please 
+look through all the categories, and choose the one that best suits), and press "Propose file change".
```
Comment (gzm0) [inline index.md]: I would omit the parenthesis. It is implied by "appropriate" and will not make lazy people less lazy.

File: index.md [hunk 3]
```diff
@@ -64,6 +64,12 @@ Some of them, like `scala-js-dom` and `scala-js-jquery`, are specific to Javascr
 Some others, such as `scala-async`, are pure-macro projects, and thus work with Scala.js out of the box.
 Most of them, though, are cross-compiling over JVM and JS, and thus have separate artifacts to run on each platform.
 
+If you have a library that would be useful to the Scala.js community, please add it to this list! 
+[This link](https://github.com/scala-js/scala-js-website/edit/master/index.md) takes you directly to the 
+editor for this page in GitHub. Add a new entry for the library under the appropriate category below (please 
+look through all the categories, and choose the one that best suits), and press "Propose file change". 
+That will submit a pull request to the owners of this site.
```
Comment (gzm0) [inline index.md]: just write: "to create a PR". Talking about the "owners of this site" on the site itself is pretty strange :)

**Comment by gzm0** [comment]
Thanks for taking initiative on this!

**Comment by jducoeur** [comment]
No prob -- how's it look now? (Added another small clause: I'd forgotten the "Create a Pull Request" step.)

**Comment by gzm0** [comment]
Looking good. However, please squash the commits and prepend "Fix #144: " to the commit message.

**Comment by jducoeur** [comment]
Hmm.  Is there any way to squash from the GitHub UI?  In general, squashing is the part that I find most problematic -- AFAIK, there's no way to do it from either GitHub, nor the GitHub for Windows client that I do all my development in.  (It really is a poorly-documented pain in the tuchus for those of us who aren't especially expert in git.  I usually wind up scrapping my fork and replacing it, because that's easier than dealing with squashing.)

**Comment by gzm0** [comment]
I am not aware of any way to do so from any UI (except for magit, if you count it as UI). Unfortunately it is the only way to have a sane commit history IMO.

**Comment by sjrd** [comment]
Crash course to squashing:
```
$ git checkout branch-to-squash
$ git rebase -i master # branch you're PRing against
```
now replace `pick` by `squash` for all but the first commit to be squashed together. Save and exit
Done.

**Comment by gzm0** [comment]
Yes... But this pretty much shatters my argument that it is easy to add stuff to the website :)

**Comment by gzm0** [comment]
So basically: It is super easy to make a PR, but then these two annoying guys with the 4 letter pseudos come and tell you to do weird git-foo :P

**Comment by sjrd** [comment]
Ah ... right. Damn 4-letter pseudos.

**Comment by jducoeur** [comment]
Yeah, 'fraid so.  Ease of use with this process is directly proportional to one's git expertise, and it's currently easy to do development with only a modest amount of that.  (This isn't the first time I've hit this issue.)  Not to mention that rebase is somewhat controversial itself -- some of the git workflows I've read assert that it's bad practice, and are designing company policies to *not* use it...

**Comment by gzm0** [comment]
So far I have not seen any git workflow that gives reasonable history and does never rebase (or squash or similarly) anywhere. Of course, you should never rebase the public branches (as in master).

**Comment by jducoeur** [comment]
Gahhh -- managed to get my repo into a bad state.  Giving up and starting again from scratch.

**Pull request closed without merging**